src/reanimated2/animation/transformationMatrix/matrixUtils.tsx

```diff
@@ -0,0 +1,433 @@
+        a[2][3] * b[3][2],
+
+      a[2][0] * b[0][3] +
+        a[2][1] * b[1][3] +
+        a[2][2] * b[2][3] +
+        a[2][3] * b[3][3],
+    ],
+    [
+      a[3][0] * b[0][0] +
+        a[3][1] * b[1][0] +
+        a[3][2] * b[2][0] +
+        a[3][3] * b[3][0],
+
+      a[3][0] * b[0][1] +
+        a[3][1] * b[1][1] +
+        a[3][2] * b[2][1] +
+        a[3][3] * b[3][1],
+
+      a[3][0] * b[0][2] +
+        a[3][1] * b[1][2] +
+        a[3][2] * b[2][2] +
+        a[3][3] * b[3][2],
+
+      a[3][0] * b[0][3] +
+        a[3][1] * b[1][3] +
+        a[3][2] * b[2][3] +
+        a[3][3] * b[3][3],
+    ],
+  ];
+}
+
+function linearOperationOnMatrix<T extends AffiniteMatrixFlat | AffiniteMatrix>(
+  _a: T,
+  func: (x: number) => number
+): T {
+  'worklet';
+
+  const isFlatOnStart = !!isAffiniteMatrixFlat(_a);
+  const a: AffiniteMatrixFlat = maybeFlattenMatrix(_a);
+
+  a.forEach((_, i) => {
+    a[i] = func(a[i]);
+  });
+
+  return isFlatOnStart ? (a as T) : (unflatten(a) as T);
+}
+
+function linearOperationOnMatrices<
+  T extends AffiniteMatrixFlat | AffiniteMatrix
+>(maybeFlatA: T, maybeFlatB: T, operation: 'add' | 'subtract'): T {
+  'worklet';
+
+  const isFlatOnStart = !!isAffiniteMatrixFlat(maybeFlatA);
+  const a: AffiniteMatrixFlat = maybeFlattenMatrix(maybeFlatA);
+  const b: AffiniteMatrixFlat = maybeFlattenMatrix(maybeFlatB);
+
+  a.forEach((_, i) => {
+    switch (operation) {
+      case 'subtract':
+        a[i] -= b[i];
+        break;
+      case 'add':
+        a[i] += b[i];
+        break;
+    }
+  });
+
+  return isFlatOnStart ? (a as T) : (unflatten(a) as T);
+}
+
+export function subtractMatrices<T extends AffiniteMatrixFlat | AffiniteMatrix>(
+  _a: T,
+  _b: T
+): T {
+  'worklet';
+
+  return linearOperationOnMatrices(_a, _b, 'subtract');
+}
+
+export function addMatrices<T extends AffiniteMatrixFlat | AffiniteMatrix>(
+  _a: T,
+  _b: T
+): T {
+  'worklet';
+
+  return linearOperationOnMatrices(_a, _b, 'add');
+}
+
+export function scaleMatrix<T extends AffiniteMatrixFlat | AffiniteMatrix>(
+  _a: T,
+  scalar: number
+): T {
+  'worklet';
+
+  return linearOperationOnMatrix(_a, (x) => x * scalar);
+}
+
+export function getRotationMatrix(
+  angle: number,
+  axis: Axis = 'z'
+): AffiniteMatrix {
+  'worklet';
+  const cos = Math.cos(angle);
+  const sin = Math.sin(angle);
+  switch (axis) {
+    case 'z':
+      return [
+        [cos, sin, 0, 0],
+        [-sin, cos, 0, 0],
+        [0, 0, 1, 0],
+        [0, 0, 0, 1],
+      ];
+    case 'y':
+      return [
+        [cos, 0, -sin, 0],
+        [0, 1, 0, 0],
+        [sin, 0, cos, 0],
+        [0, 0, 0, 1],
+      ];
+    case 'x':
+      return [
+        [1, 0, 0, 0],
+        [0, cos, sin, 0],
+        [0, -sin, cos, 0],
+        [0, 0, 0, 1],
+      ];
+  }
+}
+
+function norm(x: number, y: number, z: number) {
+  'worklet';
+  return Math.sqrt(x * x + y * y + z * z);
+}
+
+function transposeMatrix(matrix: AffiniteMatrix): AffiniteMatrix {
```

Comment:
The ones that are very likely to be delivered to the user as a part of library support both input types, the ones that should be only internal don't. Before starting further work on matrix-manipulation helpers I would like to discuss some conventions, for example it could be neat to prefix internal functions with "_"